app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java

```diff
@@ -0,0 +1,35 @@
+package com.teamagam.gimelgimel.app.mainActivity.viewmodel;
+
+import com.teamagam.gimelgimel.app.injectors.scopes.PerActivity;
+
+import java.util.ArrayList;
+
+import javax.inject.Inject;
+
+/**
+ * Created by Admin on 13/12/2016.
+ */
+
+@PerActivity
+public class PanelViewModel {
+
+    private String[] mTitles;
+    private int mNumOfTitles;
+
+
+    @Inject
+    public PanelViewModel() {}
+
+    public void setArguments (String[] titles, int numOfTitles) {
```

Comment:
Yeah. The solution to this situation is creating a factory that will inject this class' dependencies and ask for the required parameters. 
AutoFactory helps with that and generates the boilerplate code for you.
I can help you with that.